resources/vep/plugins/Capice.pm

```diff
@@ -0,0 +1,156 @@
+package Capice;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepPlugin;
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepPlugin);
+
+=head1 NAME
+ Capice
+=head1 SYNOPSIS
+ mv Capice.pm ~/.vep/Plugins
+ ./vep -i variations.vcf --plugin Capice,/FULL_PATH_TO_CAPICE_GZ_file
+
+=head1 DESCRIPTION
+ Plugin to annotate if variants with CAPICE scores and suggested classes.
+=cut
+
+sub version {
+    return '1.0';
+}
+
+sub feature_types {
+    return [ 'Transcript', 'RegulatoryFeature', 'MotifFeature', 'Intergenic'];
+}
+
+sub get_header_info {
+    return {
+        CAPICE_SC => "CAPICE score",
+        CAPICE_CL => "CAPICE classification"
+    };
+}
+
+my $self;
+
+sub new {
+    if (!(defined $self)) {
+        my $class = shift;
+        $self = $class->SUPER::new(@_);
+        my $file = $self->params->[0];
+        die("ERROR: input file not specified\n") unless $file;
+        parse_file($file);
+    }
+    return $self;
+}
+
+sub create_key {
+    my $chr = $_[0];
+    my $pos = $_[1];
+    my $ref = $_[2];
+    my $alt = $_[3];
+    my $gene = $_[4];
+    my $source = $_[5];
+    my $transcript = $_[6];
+    return "${chr}_${pos}_${ref}_${alt}_${gene}_${source}_${transcript}";
+}
+
+sub parse_file_header {
+    my @tokens = split /\t/, $_[0];
+
+    my $col_idx;
+    for my $i (0 .. $#tokens) {
+        if ($tokens[$i] eq "chr") {
+            $col_idx->{idx_chr} = $i;
+        }
+        if ($tokens[$i] eq "pos") {
+            $col_idx->{idx_pos} = $i;
+        }
+        if ($tokens[$i] eq "ref") {
+            $col_idx->{idx_ref} = $i;
+        }
+        if ($tokens[$i] eq "alt") {
+            $col_idx->{idx_alt} = $i;
+        }
+        if ($tokens[$i] eq "transcript") {
```

Comment:
off topic (requires CAPICE change): I think this column should be called `feature` or `feat` instead of `transcript` since it can contain non-transcript values as well. furthermore an additional column `feature_type` or `feat_type` should be introduced to prevent potential issues with the same identifier for different types.